src/Serializer.php

```diff
@@ -272,4 +277,24 @@ protected function getIncluded()
         $include = $this->request->get($this->expandParam);
         return is_string($include) ? preg_split('/\s*,\s*/', $include, -1, PREG_SPLIT_NO_EMPTY) : [];
     }
+
+    /**
+     * Checks Identification of resource object.
+     * http://jsonapi.org/format/#document-resource-object-identification
+     * @param ResourceInterface $resource
+     */
+    protected function checkIdentification(ResourceInterface $resource)
+    {
+        if (!is_string($resource->getId())) {
```

Comment:
@acorncom Yes i have read the specification. And i agree that `id` and `type` MUST be converted into a string. I just want to do it in another way rather then putting `(string)` everywhere where `getId()` exists. It is not flexible at all. Lets say you have created a custom client that does expect `id` as integer or you allow `id` to be null.